components/inbox-mail/CreateNewMailModal.tsx

```diff
@@ -0,0 +1,424 @@
+import useRefState from "../../hooks/useRefState";
+import { InfoButton } from "../InfoButton";
+import { MemoIconMail } from "../kern-icons/icons";
+import { User, InboxMailThread } from "./types-mail";
+import { Fragment } from 'react'
+import KernButton from "../kern-button/KernButton";
+import KernDropdown from "../KernDropdown";
+interface CreateNewMailModalProps {
+    open: boolean;
+    setOpen: (open: boolean) => void;
+    handleInboxMailCreation: (content: string, recipientIds?: string[], subject?: string, isImportant?: boolean, metaData?: any, threadId?: string) => void;
+    users: User[];
+    currentUser: User;
+    isAdmin: boolean;
+    translator: (key: string) => string;
+    selectedOrganization?: any;
+    setSelectedOrganization?: (org: any) => void;
+    organizations?: any[];
+    thread?: InboxMailThread;
+    isNewThread?: boolean;
+    isAdminSupportThread?: boolean;
+};
+
+export default function CreateNewMailModal(props: CreateNewMailModalProps) {
+    const router = useRouter();
+    const t = useMemo(() => props.translator, [props.translator]);
+
+    const projectId = router.query.projectId as string;
+    const chatId = router.query.chatId as string;
+    const { state: subject, setState: setSubject, ref: subjectRef } = useRefState('');
+    const { state: content, setState: setContent, ref: contentRef } = useRefState('');
+    const { state: includeProject, setState: setIncludeProject, ref: includeProjectRef } = useRefState(false);
+    const { state: includeChat, setState: setIncludeChat, ref: includeChatRef } = useRefState(false);
+    const { state: isImportant, setState: setIsImportant, ref: isImportantRef } = useRefState(false);
+    const { state: selectedPeople, setState: setSelectedPeople, ref: selectedPeopleRef } = useRefState<User[]>([]);
+
+
+    const cancelButtonRef = useRef(null);
+
+    useEffect(() => {
+        if (!(props.open && props.thread && !props.isNewThread && props.thread.latestMail && props.users)) return;
+
+        const currentThreadPeople = props.thread.participantIds;
+        setSelectedPeople(props.users.filter((user) => currentThreadPeople.includes(user.id) && user.id !== props.currentUser.id));
+        setSubject(props.thread.subject);
+    }, [props.open, props.thread, props.isNewThread, props.users]);
+
+    useEffect(() => {
+        if (projectId) {
+            setIncludeProject(true);
+        }
+        if (chatId) {
+            setIncludeChat(true);
+        }
+    }, [projectId, chatId]);
+
+    const initModal = useCallback(() => {
+        setSelectedPeople([]);
+        setSubject('');
+        setContent('');
+        props.setSelectedOrganization?.(null);
+        setIsImportant(false);
+    }, []);
+
+    const clearRouterParams = useCallback(() => {
+        if (!(projectId || chatId)) return;
+        router.replace('/inbox-mail');
+    }, [projectId, chatId]);
+
+    const onTransitionComplete = useCallback(initModal, []);
```

Comment:
init modal changes based on change array but this callback will always hold the initial version. also since nothing is done it's not needed (initModal can be directly set to the param)